core/service/executor.cpp

```diff
@@ -53,27 +66,81 @@ namespace executor{
         }
     }
 
+    void valueValidation(Api::Asset asset, std::vector<std::string> names){
+
+    }
+
     void transfer(const Transaction& tx){
         if(tx.has_asset()){
             // Transfer<Asset>
             auto sender = tx.senderpubkey();
             auto receiver = tx.receivepubkey();
             const auto assetName = tx.asset().name();
-            auto senderAsset    = repository::asset::find(  sender,assetName);
-            auto receiverAsset  = repository::asset::find(receiver,assetName);
+
             // **********************************************************************************
-            // * This is Transfer<Asset>'s logic.
+            // * This is Transfer<Asset>'s logic. Tax send logic
             // **********************************************************************************
+            if(tx.asset().value().find("author") != tx.asset().value().end()) {
+                const auto author = tx.asset().value().at("author").valuestring();
+
+                const auto value = tx.asset().value().at("value").valueint();
 
+                auto senderAsset = repository::asset::find(sender, assetName);
+                auto authorAsset = repository::asset::find(author, assetName);
 
+                auto receiverAsset = repository::asset::find(receiver, assetName);
+                if(senderAsset.value().find("value") != senderAsset.value().end() &&
+                   receiverAsset.value().find("value") != receiverAsset.value().end()){
+                    auto senderValue = senderAsset.value().at("value").valueint();
+                    auto receiverValue = receiverAsset.value().at("value").valueint();
+                    if(senderValue > value){
+                        (*senderAsset.mutable_value())["value"].set_valueint(senderValue - value);
+                        (*receiverAsset.mutable_value())["value"].set_valueint(receiverValue + value);
+                        (*authorAsset.mutable_value())["value"].set_valueint(value * 0.8);
+                    }
+                }
+                repository::asset::update(sender, assetName, senderAsset);
+                repository::asset::update(receiver, assetName, receiverAsset);
+
+            // **********************************************************************************
+            // * This is Transfer<Asset>'s logic. multi message chat
             // **********************************************************************************
-            repository::asset::update(sender,assetName,senderAsset);
-            repository::asset::update(receiver,assetName,receiverAsset);
+            }else if(tx.asset().value().find("targetName") != tx.asset().value().end()){
```

Comment:
this else-if branch are same
Extract that to a separate function and pass different strings so there won't code duplicates